stampede.js

```diff
@@ -129,7 +130,17 @@ class Stampede {
       payload.data = d;
       payload.__caching__ = false;
       if (d && d.error) payload.error = true;
+
+      let whenExecutedPromise;
+      if (this.whenFnExecuted) {
+        whenExecutedPromise = this.whenFnExecuted(key, payload);
```

Comment:
If this errors synchronously for some reason then the data is lost.

Maybe it might be safer to switch the order?
i.e.
```
let updatedPromise = this._adapter.update(key,payload,expiry);
if (this.whenFnExecuted) {
  await this.whenExecuted(key, payload)
}

await updatedPromise;

payload.data = raw_data
...
```

Reply:
oh interesting.